tests/test_utils.py

```diff
@@ -3,102 +3,145 @@
+
 
 def test_pattern_match_none():
     data = pd.Series(['foo', 'bar'])
     values = ['baz']
 
     obs = utils.pattern_match(data, values)
-    exp = [False, False]
-
-    assert (obs == exp).all()
+    assert (obs == [False, False]).all()
 
 
 def test_pattern_match_nan():
     data = pd.Series(['foo', np.nan])
     values = ['baz']
 
     obs = utils.pattern_match(data, values)
-    exp = [False, False]
-
-    assert (obs == exp).all()
+    assert (obs == [False, False]).all()
 
 
 def test_pattern_match_one():
     data = pd.Series(['foo', 'bar'])
     values = ['foo']
 
     obs = utils.pattern_match(data, values)
-    exp = [True, False]
-
-    assert (obs == exp).all()
+    assert (obs == [True, False]).all()
 
 
 def test_pattern_match_str_regex():
     data = pd.Series(['foo', 'foo2', 'bar'])
     values = ['foo']
 
     obs = utils.pattern_match(data, values)
-    exp = [True, False, False]
-
-    assert (obs == exp).all()
+    assert (obs == [True, False, False]).all()
 
 
 def test_pattern_match_ast_regex():
     data = pd.Series(['foo', 'foo2', 'bar'])
     values = ['foo*']
 
     obs = utils.pattern_match(data, values)
-    exp = [True, True, False]
+    assert (obs == [True, True, False]).all()
+
 
-    assert (obs == exp).all()
+def test_pattern_match_ast2_regex():
+    data = pd.Series(['foo|bar', 'foo', 'bar'])
+    values = ['*o*b*']
+
+    obs = utils.pattern_match(data, values)
+    assert (obs == [True, False, False]).all()
 
 
 def test_pattern_match_plus():
     data = pd.Series(['foo', 'foo+', '+bar', 'b+az'])
     values = ['*+*']
 
     obs = utils.pattern_match(data, values)
-    exp = [False, True, True, True]
-
-    assert (obs == exp).all()
+    assert (obs == [False, True, True, True]).all()
 
 
 def test_pattern_match_dot():
     data = pd.Series(['foo', 'fo.'])
     values = ['fo.']
 
     obs = utils.pattern_match(data, values)
-    exp = [False, True]
-
-    assert (obs == exp).all()
+    assert (obs == [False, True]).all()
 
 
 def test_pattern_match_brackets():
     data = pd.Series(['foo (bar)', 'foo bar'])
     values = ['foo (bar)']
 
     obs = utils.pattern_match(data, values)
-    exp = [True, False]
-
-    assert (obs == exp).all()
+    assert (obs == [True, False]).all()
 
 
 def test_pattern_match_dollar():
     data = pd.Series(['foo$bar', 'foo'])
     values = ['foo$bar']
 
     obs = utils.pattern_match(data, values)
-    exp = [True, False]
-
-    assert (obs == exp).all()
+    assert (obs == [True, False]).all()
 
 
 def test_pattern_regexp():
     data = pd.Series(['foo', 'foa', 'foo$'])
     values = ['fo.$']
 
     obs = utils.pattern_match(data, values, regexp=True)
-    exp = [True, True, False]
+    assert (obs == [True, True, False]).all()
+
+
+def test_find_depth():
+    obs = utils.find_depth(TEST_VARS)
+    assert obs == [0, 1, 2]
+
+
+def test_find_depth_with_str():
+    data = pd.Series(['foo', 'foo|bar|baz', 'bar|baz', 'bar|baz|foo'])
+    obs = utils.find_depth(data, 'bar')
```

Comment:
for this (and below) tests, can you please also add explicit kwargs. Without them, it's hard to know which arguments are actually being used (especially when you look months later). So for this, it would be

```
obs = utils.find_depth(data, s='bar')
```

Sorry for so many comments, but I think I'm done!